External/FEXCore/Source/Interface/Core/GdbServer.cpp

```diff
@@ -80,9 +80,9 @@ GdbServer::GdbServer(FEXCore::Context::Context *ctx) : CTX(ctx) {
 
   // This is a total hack as there is currently no way to resume once hitting a segfault
   // But it's semi-useful for debugging.
-  for (uint32_t Signal = 0; Signal <= SignalDelegator::MAX_SIGNALS; ++Signal) {
+  for (uint32_t Signal = 0; Signal < SignalDelegator::MAX_SIGNALS; ++Signal) {
     ctx->SignalDelegation->RegisterHostSignalHandler(Signal, [this] (FEXCore::Core::InternalThreadState *Thread, int Signal, void *info, void *ucontext) {
-      if (PassSignals[Signal]) {
+      if (PassSignals[Signal - 1]) {
```

Comment:
The variable shadowing of `Signal` is rather confusing here (it looks like this line would index using `-1` on first iteration, but it isn't doing that since the lambda introduces a shadowed `Signal` variable that starts at 1)

I'd suggest running the loop from 1 to `MAX_SIGNALS` (inclusive) instead, so that all `Signal` variables have the same semantics.